include/llbuild/Core/BuildEngine.h

```diff
@@ -269,6 +272,21 @@ class BuildEngine {
   /// discovered currently.
   const ValueType& build(const KeyType& key);
 
+  /// Cancel the currently running build.
+  ///
+  /// The engine guarantees that it will not *start* any task after it has been
```

Comment:
Good point. It probably wouldn't be much overhead to check it via an atomic (since it is almost never written), but I agree doesn't seem worth it currently. Updated comment in 5ab21ef8fe6566cf9dee4d7d9b2be3eab9b4ec08